src/wallet/wallet.cpp

```diff
@@ -4042,11 +4042,17 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
     m_internal_spk_managers.clear();
 
     // Setup new descriptors
-    SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+    SetWalletFlag(WALLET_FLAG_DESCRIPTORS | WALLET_FLAG_GLOBAL_HD_KEY);
     if (!IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
         // Use the existing master key if we have it
         if (data.master_key.key.IsValid()) {
             SetupDescriptorScriptPubKeyMans(data.master_key);
+
+            // Store the master key as our active hd key
+            if (!AddHDKey(data.master_key)) {
+                throw std::runtime_error(std::string(__func__) + ": Unable to add original seed's HD key");
+            }
+            SetActiveHDKey(data.master_key.Neuter());
```

Comment:
In 283fe566d:

Should check the error here.
```c++
if (!SetActiveHDKey(data.master_key.Neuter())) {
    error += _("Error: Unable to set active HD key");
    return false;
}
```

Reply:
Done as suggested